components/template/vue/vue-template.spec.ts

```diff
@@ -1,41 +1,22 @@
 import { makeTemplate } from './vue-template';
 import { expect } from 'chai';
+import vueIndex from './vueIndex';
 
 it('should create default code', () => {
 	const result = makeTemplate({ npmId: '@bit/bit.javascript.raw.code-generator' });
 
-	const indexvue = [
-		'<template>',
-		'	<CodeGenerator/>',
-		'</template>',
-		'',
-		'<script>',
-		"	import Vue from 'vue';",
-		"	import CodeGenerator from '@bit/bit.javascript.raw.code-generator';",
-		'	',
-		'	export default (',
-		'		{',
-		'			data: () => ({',
-		'				var1: "world"',
-		'			}),',
-		'			components: {',
-		'				CodeGenerator,',
-		'			}',
-		'		}',
-		'	)',
-		'</script>',
-		'',
-		'<style scoped>',
-		'',
-		'</style>',
-	].join('\n');
+	result.files['Index.vue'] = result.files['Index.vue']
+		.replace(/\t/g, '').replace(/  /g, ''); // tabs and double-spaces removed for comparison
```

Comment:
you are hiding bugs!
I know it is difficult to compare with whitespaces, but this is what actually goes on the screen. 

Here is what I got when I ran it locally:
### expected:
```vue
<template>
  <CodeGenerator/>
</template>

<script>
  import CodeGenerator from '@bit/bit.javascript.raw.code-generator';

  export default (
    {
      data () {
        return {
          var1: 'World'
        }
      },
      components: {
        CodeGenerator
      }
    }
  )
</script>

<style scoped>

</style>
```

### actual
```vue
<template>
                <CodeGenerator/>
        </template>

        <script>
                import CodeGenerator from '@bit/bit.javascript.raw.code-generator';

        export default (
                {
                                data () {
                                        return {
                                                var1: 'World'
                                        }
                                },
                                components: {
                                        CodeGenerator
                                }
                        }
        )
        </script>

        <style scoped>

        </style>
```

Reply:
you can see it more easily by adding ```console.log()```

Reply:
It is difficult to compare with whitespaces, since different environments appear to interpret spaces/tabs differently in different files. My tests worked and console logging output (before replacing anything) showed \t in place of tabs from the generated code, but spaces in the snapshot code.

In the new commit, I've only replaced spaces in the snapshot with tabs (expecting tabs in the generated code to be correct without modification.

I suggest that what should actually happen is that the code is generated without tabs and compared to a snapshot without tabs, then prettified by the editor at runtime (e.g. the example editor on bit.dev should put the tabs in, as would any personal editor based on the editor's settings for the number of spaces per tab). Since tabs/spaces are not required for javascript to be interpreted, they should not be included in code used for testing.

Reply:
looks good. I cloned it locally and the test pass. It also worked when I inlined the tabs.

As for prettifier, yes, it looks like the correct way to go, but good luck getting a light vue prettifier that works in a browser! haha.